features/nationaliteithistorie/dev/nationaliteit-gegevens.feature

```diff
@@ -0,0 +1,129 @@
+# language: nl
+
+Functionaliteit: Lever de juiste gegevens over een nationaliteit
+
+  Als consumer van de BRP
+  Wil ik de nationaliteiten van een persoon met alle geldige gegevens
+
+  In de bron worden gegevens over een nationaliteit of bijzonder Nederlanderschap opgeslagen in één categorievoorkomen met de actuele gegevens (categorie 4) en optioneel in een of meerdere categorievoorkomens met "historische" gegevens (categorie 54).
+
+  Voor een nationaliteit die is beëindigd, is de "actuele" situatie dat er geen nationaliteit meer is, waardoor de gegevens over die (beëindigde) nationaliteit niet "actueel" zijn en dus in historische categorievoorkomen(s) staan.
+  Alleen gegevens over het beëindigen van de nationaliteit staan dan in het "actuele" categorievoorkomen.
+
+  Rule: Voor een niet-beëindigde nationaliteit of bijzonder Nederlanderschap wordt de datumIngangGeldigheid gevuld met de datum geldigheid (85.10) in de actuele categorie (04).
+    - een nationaliteit of bijzonder Nederlanderschap is niet beëindigd wanneer reden beëindigen (64.10) leeg is of geen waarde heeft.
+
+    Scenario: persoon heeft een niet-beëindigde nationaliteit
+      Gegeven de persoon met burgerservicenummer '000009830' heeft een 'nationaliteit' met de volgende gegevens
+      | nationaliteit (05.10) | reden opnemen (63.10) | datum ingang geldigheid (85.10) |
+      | 0001                  | 001                   | 19750707                        |
+      Als nationaliteithistorie wordt geraadpleegd met de volgende parameters
+      | naam                | waarde                                                                                                    |
+      | type                | RaadpleegMetPeildatum                                                                                     |
+      | burgerservicenummer | 000009830                                                                                                 |
+      | peildatum           | 2022-08-16                                                                                                |
+      | fields              | nationaliteiten.nationaliteit.code,nationaliteiten.redenOpname.code,nationaliteiten.datumIngangGeldigheid |
+      Dan heeft de response de volgende 'nationaliteiten'
+      | type          | nationaliteit.code | redenOpname.code | datumIngangGeldigheid.type | datumIngangGeldigheid.datum | datumIngangGeldigheid.langFormaat |
+      | Nationaliteit | 0001               | 001              | Datum                      | 1975-07-07                  | 7 juli 1975                       |
+
```

Comment:
Is het niet beter om in de dev scenario's maar één veld tegelijk te valideren? Nu kan deze scenario om drie redenen falen, namelijk 'nationaliteit' en/of 'reden opnemen' en/of 'datum ingang geldigheid' worden niet goed gemapt. Deze scenario kan je als volgt beschrijven:

    Abstract Scenario: persoon heeft een niet-beëindigde nationaliteit
      Gegeven de persoon met burgerservicenummer '000009830' heeft een 'nationaliteit' met de volgende gegevens
      | nationaliteit (05.10) | reden opnemen (63.10) | datum ingang geldigheid (85.10) |
      | 0001                  | 001                   | 19750707                        |
      Als nationaliteithistorie wordt geraadpleegd met de volgende parameters
      | naam                | waarde                |
      | type                | RaadpleegMetPeildatum |
      | burgerservicenummer | 000009830             |
      | peildatum           | 2022-08-16            |
      | fields              | <field>               |
      Dan heeft de response de volgende 'nationaliteiten'
      | type          | <naam>   |
      | Nationaliteit | <waarde> |

      Voorbeelden:
      | field                              | naam               | waarde |
      | nationaliteiten.nationaliteit.code | nationaliteit.code | 0001   |
      | nationaliteiten.redenOpname.code   | redenOpname.code   | 001    |

    Scenario: persoon heeft een niet-beëindigde nationaliteit
      Gegeven de persoon met burgerservicenummer '000009830' heeft een 'nationaliteit' met de volgende gegevens
      | nationaliteit (05.10) | reden opnemen (63.10) | datum ingang geldigheid (85.10) |
      | 0001                  | 001                   | 19750707                        |
      Als nationaliteithistorie wordt geraadpleegd met de volgende parameters
      | naam                | waarde                                |
      | type                | RaadpleegMetPeildatum                 |
      | burgerservicenummer | 000009830                             |
      | peildatum           | 2022-08-16                            |
      | fields              | nationaliteiten.datumIngangGeldigheid |
      Dan heeft de response de volgende 'nationaliteiten'
      | type          | datumIngangGeldigheid.type | datumIngangGeldigheid.datum | datumIngangGeldigheid.langFormaat |
      | Nationaliteit | Datum                      | 1975-07-07                  | 7 juli 1975                       |

Reply:
Ik zit waarschijnlijk niet in de doelgroep voor deze vraag, maar waarom is dat beter ? Heeft dat met geautomatiseerd testen te maken ? Ik wil niet zozeer een discussie starten, maar het vooral begrijpen. Als dat teveel effort kost, gewoon mag je deze opmerking ook negeren.

Reply:
ik ben nog verder gegaan en heb per veld een eigen feature gemaakt. Dan kan het ook per veld geïmplementeerd en getest worden.

Alleen nationaliteiten.type zit door elke feature, omdat die verplicht geleverd wordt.

Reply:
> Ik zit waarschijnlijk niet in de doelgroep voor deze vraag, maar waarom is dat beter ? Heeft dat met geautomatiseerd testen te maken ? Ik wil niet zozeer een discussie starten, maar het vooral begrijpen. Als dat teveel effort kost, gewoon mag je deze opmerking ook negeren.

In de scenario worden meerdere velden gevalideerd. Als de scenario faalt, moet er worden uitgezocht (ook al is het minimale check) bij welke van de velden het fout gaat en wat eigenlijk de fout veroorzaakt. Is het de mapping? Of de parameters in de Als stap? Wordt er maar één veld gevraagd en gevalideerd, dan is sneller te zien waar de fout zit.